### PR TITLE
feat: data fetching UI

### PR DESCRIPTION
**Pull request by peelar**
I would appreciate your feedback.

The task was to add an example of data fetching the App Template. I added it alongside the current `DashboardActions` (to have one view with all the app capabilities). The issue is, the `DashboardActions` are appended to the rest of the `pages/index`, when the app is installed in the dashboard. With the new content inside `DashboardActions`, the index page is cluttered (especially considering new content is coming in this PR https://github.com/saleor/saleor-app-template/pull/175). 

So I decided not to import `DashboardActions` in `pages/index`, and turn it into a separate page `/actions`:

image

Which makes the index page looks like this:

image

where the "See what your app can do" link leads to `/actions`. This way, we have one page for educational content and one page with feature examples.

But the "See what your app can do" link looks kind of unimpressive. Do you have any idea what we can do to make it pop more? Do you have any feedback about this approach?

### CODE REVIEW

File: graphql/queries/Orders.graphql [hunk 1]
```diff
@@ -0,0 +1,16 @@
+query Orders {
```
Comment (krzysztofwolski) [inline graphql/queries/Orders.graphql:1]: I would instead use here different name for this query, for example `RecentOrders`. There are probably already a few classes called `Orders` generated by codegen, which might confuse first-time GQL users. Giving a meaningful name for a query is also a good practice.

File: src/pages/actions.tsx [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+/**
+ * This is example of using AppBridge, when App is mounted in Dashboard
+ * See more about AppBridge possibilities
+ * https://github.com/saleor/saleor-app-sdk/blob/main/docs/app-bridge.md
+ *
+ * -> You can safely remove this file!
+ */
+const ActionsPage = () => {
+  const { appBridge, appBridgeState } = useAppBridge();
+  const [{ data, fetching }] = useOrdersQuery();
```
Comment (krzysztofwolski) [inline src/pages/actions.tsx:14]: Let's move the orders block to a separate component; this component becomes a bit too crowded to learn from. All GQL fetch handling (hook, loading state, etc) will be in one place.

I would also consider putting query inside the new component (template is already configured to do that). But that's optional.

// src/OrdersExample.tsx

```

gql`
query RecentOrders{
 orders(first: 10) {
    edges {
      node {
        id
        number
        status
        created
        user {
          firstName
          lastName
        }
      }
    }
  }
}
`

const OrdersExample = () => {
  const { appBridge, appBridgeState } = useAppBridge();
  const [{ data, fetching }] = useOrdersQuery();

  if(fetching){
    return <Text>Loading order data...</Text>
  }

  return (
  <>
      <Text color="textNeutralSubdued">
        💡 You can modify the query in the <code>graphql/queries/Orders.graphql</code> file.
        Remember to run <code>pnpm codegen</code> afterwards to regenerate the types.
      </Text>
      <ul style={{ listStyle: "inside", margin: 0 }}>
        {data?.orders?.edges?.map((order) => (
           <li key={order.node.id}>
             <a
               onClick={() => navigateToOrder(order.node.id)}
               href={`https://pikle.eu.saleor.cloud/dashboard/orders/${order.node.id}`}
                    >{`Order ${order.node.number}`}</a>{" "}
                    {`is ${order.node.status}`}
           </li>
         ))}
       </ul>
     </>
  
  )
}
```

**Review by krzysztofwolski — CHANGES_REQUESTED** [review]

File: src/pages/actions.tsx (hunk 2 above)
Comment (lkostrowski) [reply src/pages/actions.tsx:14]: agree to extract component

File: src/pages/actions.tsx [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+
+  const navigateToOrders = () => {
+    appBridge?.dispatch({
+      type: "redirect",
+      payload: {
+        to: "/orders",
+        actionId: "message-from-app",
+      },
```
Comment (lkostrowski) [inline src/pages/actions.tsx:22]: we can import `Actions.createRedirectAction` (AFAIR) from SDK that will
1. show that app-sdk can do that
2. hide actionId which is implementation detail

File: src/pages/actions.tsx [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+        </Text>
+
+        <>
+          {fetching ? (
+            <Text>Fetching orders...</Text>
+          ) : (
+            <>
+              <Text color="textNeutralSubdued">
+                💡 You can modify the query in the <code>graphql/queries/Orders.graphql</code> file.
+                Remember to run <code>pnpm codegen</code> afterwards to regenerate the types.
```
Comment (lkostrowski) [inline src/pages/actions.tsx:85]: ` Remember to run <code>pnpm codegen</code> script`

File: src/pages/actions.tsx [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+              </Text>
+              <ul style={{ listStyle: "inside", margin: 0 }}>
+                {data?.orders?.edges?.map((order) => (
+                  <li key={order.node.id}>
+                    <a
+                      onClick={() => navigateToOrder(order.node.id)}
+                      href={`https://pikle.eu.saleor.cloud/dashboard/orders/${order.node.id}`}
```
Comment (lkostrowski) [inline src/pages/actions.tsx:92]: hardcoded env

File: src/pages/index.tsx [hunk 6]
```diff
@@ -65,9 +41,12 @@ const IndexPage: NextPage = () => {
   return (
     <Box padding={8}>
       <Text variant={"hero"}>Welcome to Saleor App Template (Next.js) 🚀</Text>
-      <Text as={"p"} marginTop={8}>
-        This is a boilerplate you can start with, to create an app connected to Saleor
+      <Text as={"p"} marginY={4}>
+        Saleor App Template is a minimalistic boilerplate that provides a working example of a
+        Saleor app.
       </Text>
+      {appBridgeState?.ready && mounted && <Link href="/actions">See what your app can do →</Link>}
```
Comment (lkostrowski) [inline src/pages/index.tsx:48]: maybe use Macaw UI Button to make it more impressive?

**Review by lkostrowski — CHANGES_REQUESTED** [review]

File: src/pages/actions.tsx (hunk 5 above)
Comment (peelar) [reply src/pages/actions.tsx:92]: lol, good catch

**Comment by peelar** [comment]
@krzysztofwolski @lkostrowski addressed all

**Review by lkostrowski — APPROVED** [review]

**Review by krzysztofwolski — APPROVED** [review]
👍

**Pull request merged**